datumaro/plugins/synthetic_images_plugin/utils.py

```diff
@@ -0,0 +1,162 @@
+    model_name = 'colorization_release_v2.caffemodel'
+    npy_name = 'pts_in_hull.npy'
+
+    if not os.path.exists(os.path.join(path, proto_name)):
+        url = 'https://raw.githubusercontent.com/richzhang/colorization/caffe/colorization/models/'
+        proto = requests.get(url + proto_name)
+        open(os.path.join(path, proto_name), 'wb').write(proto.content)
+    if not os.path.exists(os.path.join(path, model_name)):
+        url = 'http://eecs.berkeley.edu/~rich.zhang/projects/2016_colorization/files/demo_v2/'
+        model = requests.get(url + model_name)
+        open(os.path.join(path, model_name), 'wb').write(model.content)
+    if not os.path.exists(os.path.join(path, npy_name)):
+        url = 'https://github.com/richzhang/colorization/raw/caffe/colorization/resources/'
+        pts_in_hull = requests.get(url + npy_name)
+        open(os.path.join(path, npy_name), 'wb').write(pts_in_hull.content)
+
+
+def rgb2lab(frame):
+    y_coeffs = np.array([0.212671, 0.715160, 0.072169], dtype=np.float32)
+    frame = np.where(frame > 0.04045, np.power((frame + 0.055) / 1.055, 2.4), frame / 12.92)
+    y = frame @ y_coeffs.T
+    L = np.where(y > 0.008856, 116 * np.cbrt(y) - 16, 903.3 * y)
+    return L
+
+
+def colorize(frame, net):
+    H_orig, W_orig = frame.shape[:2] # original image size
+    if len(frame.shape) == 2 or frame.shape[-1] == 1:
+        frame = np.tile(frame.reshape(H_orig, W_orig, 1), (1, 1, 3))
+
+    frame = frame.astype(np.float32) / 255
+    img_l = rgb2lab(frame) # get L from Lab image
+    img_rs = cv.resize(img_l, (224, 224)) # resize image to network input size
+    img_l_rs = img_rs - 50  # subtract 50 for mean-centering
+
+    net.setInput(cv.dnn.blobFromImage(img_l_rs))
+    ab_dec = net.forward()[0, :, :, :].transpose((1, 2, 0))
+
+    ab_dec_us = cv.resize(ab_dec, (W_orig, H_orig))
+    img_lab_out = np.concatenate((img_l[..., np.newaxis], ab_dec_us), axis=2) # concatenate with original image L
+    img_bgr_out = np.clip(cv.cvtColor(img_lab_out, cv.COLOR_Lab2BGR), 0, 1)
+    frame_normed = 255 * (img_bgr_out - img_bgr_out.min()) / (img_bgr_out.max() - img_bgr_out.min())
+    frame_normed = np.array(frame_normed, dtype=np.uint8)
+    return cv.resize(frame_normed, (W_orig, H_orig))
+
+
+def augment(image):
+    if np.random.random(1) >= 0.5:
+        image = cv.flip(image, 1)
+
+    if np.random.random(1) >= 0.5:
+        image = cv.flip(image, 0)
+
+    height, width = image.shape[:2]
+    angle = np.random.uniform(-30, 30)
+    rotate_matrix = cv.getRotationMatrix2D(center=(width / 2, height / 2), angle=angle, scale=1)
+    image = cv.warpAffine(src=image, M=rotate_matrix, dsize=(width, height))
+
+    image = fill_background(image)
+
+    k_size = np.random.choice(list(range(1, 16, 2)))
+    image = cv.GaussianBlur(image, (k_size, k_size), 0)
+    return image
+
+
+def fill_background(image):
+    synthetic_background = Path(__file__).parent / 'synthetic_background.npy'
```

Comment:
1. Data files that are part of the project should be loaded with `importlib.metadata`.
2. What is the origin of this file?

Reply:
Sorry, I just realized I named the wrong module in item 1. It should have been `importlib.resources`.

Reply:
Done

Reply:
You didn't answer my question, though.

Reply:
Sorry, I generated this file using the mean value by class for Imagenet images per channel